app/assets/javascripts/services/post_service.js

```diff
@@ -6,18 +6,25 @@ ManageIQ.angular.app.service('postService', ["miqService", "$timeout", "$window"
       angular.toJson({
         action: "edit",
         resource: updateObject
-      })).then(handleSuccess, handleFailure);
+      })).then(function(response) {
+        if (response.error) {
+          handleFailure(response);
+        } else {
+          handleSuccess;
+        }
+      });
```

Comment:
@h-kataria Can you leverage `handleFailure` function in `miqService` here? 
Check how it's being used in the other controllers.